wallet/src/de/schildbach/wallet/security/ModernEncryptionProvider.kt

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2023 Dash Core Group.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package de.schildbach.wallet.security
+
+import android.content.SharedPreferences
+import android.security.keystore.KeyGenParameterSpec
+import android.security.keystore.KeyProperties
+import android.util.Base64
+import org.dash.wallet.common.util.security.EncryptionProvider
+import java.nio.charset.StandardCharsets
+import java.security.GeneralSecurityException
+import java.security.KeyStore
+import java.security.KeyStoreException
+import javax.crypto.Cipher
+import javax.crypto.KeyGenerator
+import javax.crypto.SecretKey
+import javax.crypto.spec.GCMParameterSpec
+
+class ModernEncryptionProvider(
```

Comment:
I wanted to have some unit tests for this class.  This class is a very critical part of the wallet as it controls encrypting and decrypting of the wallet keys.  I will make a separate ticket for this.

While this class is identical in functionality, it makes me feel better if we had some tests that verify that the functionality is identical beyond what QA does.  I am not sure if we can easily use OpenSSL to create such tests or if we need to use the java implementation on the master branch.

Reply:
Yes, good idea. We can keep the java class around for tests.

Some Explore tests are failing in this PR, it seems like it will take time to fix them. We can do it in the same story (or separate, doesn't matter).